src/Nethermind/Nethermind.TxPool/Collections/TxDistinctSortedPool.cs

```diff
@@ -40,7 +40,7 @@ public TxDistinctSortedPool(int capacity, IComparer<Transaction> comparer, ILogM
         protected override IComparer<Transaction> GetGroupComparer(IComparer<Transaction> comparer) => comparer.GetPoolUniqueTxComparerByNonce();
         protected override IComparer<Transaction> GetReplacementComparer(IComparer<Transaction> comparer) => comparer.GetReplacementComparer();
 
-        protected override Address? MapToGroup(Transaction value) => value.MapTxToGroup();
+        protected override Address MapToGroup(Transaction value) => value.MapTxToGroup() ?? throw new NullReferenceException();
```

Comment:
Don't throw `NullReferenceException` it will confuse us if this is access to a `null` variable or some expected error. `ArgumentException` or `ArgumentNullException` is what you want.